x-pack/plugins/lens/public/datasources/form_based/datapanel.test.tsx

```diff
@@ -137,407 +157,222 @@ const fieldsTwo = [
+    spec: {},
   },
-  {
-    name: 'source',
-    displayName: 'source',
-    type: 'string',
-    aggregatable: true,
-    searchable: true,
+  '2': {
+    id: '2',
+    title: 'idx2',
+    timeFieldName: 'timestamp',
+    hasRestrictions: true,
+    fields: fieldsTwo,
+    getFieldByName: getFieldByNameFactory(fieldsTwo),
+    isPersisted: true,
+    spec: {},
   },
-  documentField,
-];
+};
 
-jest.spyOn(UseExistingFieldsApi, 'useExistingFieldsFetcher');
-jest.spyOn(UseExistingFieldsApi, 'useExistingFieldsReader');
-jest.spyOn(ExistingFieldsServiceApi, 'loadFieldExisting').mockImplementation(async () => ({
-  indexPatternTitle: 'test',
-  existingFieldNames: [],
-}));
+const core = coreMock.createStart();
+const dataViews = dataViewPluginMocks.createStartContract();
 
-const initialState: FormBasedPrivateState = {
-  currentIndexPatternId: '1',
+const constructState = (currentIndexPatternId: string): FormBasedPrivateState => ({
+  currentIndexPatternId,
   layers: {
-    first: {
-      indexPatternId: '1',
-      columnOrder: ['col1', 'col2'],
-      columns: {
-        col1: {
-          label: 'My Op',
-          dataType: 'string',
-          isBucketed: true,
-          operationType: 'terms',
-          sourceField: 'source',
-          params: {
-            size: 5,
-            orderDirection: 'asc',
-            orderBy: {
-              type: 'alphabetical',
-            },
-          },
-        } as TermsIndexPatternColumn,
-        col2: {
-          label: 'My Op',
-          dataType: 'number',
-          isBucketed: false,
-          operationType: 'average',
-          sourceField: 'memory',
-        },
-      },
-    },
-    second: {
-      indexPatternId: '1',
-      columnOrder: ['col1', 'col2'],
-      columns: {
-        col1: {
-          label: 'My Op',
-          dataType: 'string',
-          isBucketed: true,
-          operationType: 'terms',
-          sourceField: 'source',
-          params: {
-            size: 5,
-            orderDirection: 'asc',
-            orderBy: {
-              type: 'alphabetical',
-            },
-          },
-        } as TermsIndexPatternColumn,
-        col2: {
-          label: 'My Op',
-          dataType: 'number',
-          isBucketed: false,
-          operationType: 'average',
-          sourceField: 'bytes',
-        },
-      },
+    1: {
+      indexPatternId: currentIndexPatternId,
+      columnOrder: [],
+      columns: {},
     },
   },
+});
+
+const frame = getFrameAPIMock({ indexPatterns });
+const defaultProps = {
+  data: dataPluginMock.createStartContract(),
+  dataViews,
+  fieldFormats: fieldFormatsServiceMock.createStartContract(),
+  indexPatternFieldEditor: indexPatternFieldEditorPluginMock.createStartContract(),
+  indexPatternService: createIndexPatternServiceMock({
+    updateIndexPatterns: jest.fn(),
+    core,
+    dataViews,
+  }),
+  onIndexPatternRefresh: jest.fn(),
+  core,
+  dateRange: { fromDate: '2019-01-01', toDate: '2020-01-01' },
```

Comment:
why absolute dates?

Reply:
because one of the tests were using these dates, whereas the previous ones were not used at all.